Documentation review

OnorioCatenacci opened the pull request.

Commit:
Issue #5
https://github.com/christhekeele/matcha/issues/5

This is a very simple start on adding benchmarking tests to matcha.  I didn't want to add too many tests without running this by Chris Keele to insure I'm headed in the right direction.

OnorioCatenacci committed:
Merge branch 'christhekeele:latest' into latest

OnorioCatenacci committed:
Remove code

I removed an alias in the benchmark unit tests which was causing a warning.

OnorioCatenacci committed:
Merge branch 'christhekeele:latest' into latest

christhekeele commented on docs/guides/usage/tracing.livemd:
````diff
@@ -83,7 +120,7 @@ Integer.parse("3")
 <!-- livebook:{"output":true} -->
 
 ```
-Matcha.Trace: `Elixir.Integer.parse("1")` called on #PID<0.142.0>
````
This pid will change every time, so if the content of the pid changes between runs of the example, I'd rather ignore those diffs.

christhekeele commented on docs/guides/usage/tracing.livemd:
````diff
@@ -111,8 +148,9 @@ Integer.parse("1")
 <!-- livebook:{"output":true} -->
 
 ```
-Matcha.Trace: `Elixir.Integer.parse("1")` called on #PID<0.142.0>
````
The `Integer.__info__(:deprecated)` stuff is happening for some reason I don't understand; it wouldn't get called in normal code execution and spoils the examples. I think it's some iex-mode-only introspection thing?

christhekeele commented on docs/guides/usage/tracing.livemd:
````diff
@@ -165,14 +210,13 @@ Integer.parse("1")
 
 ```
-{1, ""}
+Matcha.Trace: `Elixir.Integer.parse("1")` called on #PID<0.135.0>
````
I think part of these diffs are because each execution of the cell has a chance to return (`{1, ""}`) ***and*** receive trace messages in any order. I'd generally ignore diffs where that's the only thing happening, and pretend it just always shows the return before the trace. I'd accept manual editing of the result into a stand form?

christhekeele commented on docs/guides/usage/tracing.livemd:
````diff
@@ -190,6 +234,7 @@ For example, most `Integer` functions take an optional `base` argument, defaulti
 If we are only interested in cases where we are parsing a string into an integer with the non-default decimal base, we can trace calls to `Integer.parse` with an arity of `2`—that is, functions were we are specifying an extra `base` argument:
 
 ```elixir
+# NB we're invoking calls function here.
````
What does NB mean in this context? Can we expand the acronym to avoid assumptions about the reader's familiarity with slang?

OnorioCatenacci replied:
I fixed this one Chris.

christhekeele commented on docs/guides/usage/tracing.livemd:
```diff
@@ -200,21 +245,21 @@ Integer.parse("123", 2)
 ## Tracing Specific Arguments
 
-Elixir also supports [**_function overloading_**](https://en.wikipedia.org/wiki/Function_overloading), providing different implementations for a function depending on the _specific kinds_ of arguments it is invoked with. In many languages, this is acheived base on a type system; in Elixir, we use pattern matching in the function head and guards to steer our function calls to a specific implementa ...
+Elixir also supports [**_function overloading_**](https://en.wikipedia.org/wiki/Function_overloading), providing different implementations for a function depending on the _specific kinds_ of arguments it is invoked with. In many languages, this is acheived based on a type system; in Elixir, we use pattern matching in the function head and guards to steer our function calls to a specific implement ...
```
💜

christhekeele commented on test/unit/benchmark_unit_test.exs:
```diff
@@ -0,0 +1,55 @@
+defmodule BenchmarkUnitTest do
```
I think all of this has been folded into `latest` via `test/benchmarks`, no? Might have accidentally commited from your earlier changes in #28 ?

OnorioCatenacci replied:
Yes I think that's what happened!  I'll straighten this out and either update this PR or drop it and submit a new one!

christhekeele commented on docs/guides/usage/tracing.livemd:
```diff
@@ -12,7 +12,25 @@ IO.puts("Installed matcha version: #{Application.spec(:matcha, :vsn)}")
+Generated recon app
+==> matcha
+Compiling 17 files (.ex)
+Generated matcha app
```
I think this output is always going to change between runs, which is why I left the version status report below as the only truncated output.

OnorioCatenacci committed:
Documentation Review

I fixed what looks like a typo to me and I made a small note on one of the examples to (I hope) make a minor improvement.

OnorioCatenacci commented:
Since this doesn't seem to be updating with my modified code I'm going to close this PR and submit a new one.
